tests/kernel/mem_protect/mem_map/src/main.c

```diff
@@ -12,6 +12,13 @@
 #define SKIP_EXECUTE_TESTS
 #endif
 
+/* The execution case in qemu_x86_64 is always failed when running with
+ * coverage option, so skip it temporarily.
```

Comment:
x86_64 has the NX bit, there is no technical reason to disable the execution tests.
Where is the bug on this? Have you made any attempt to root-cause this?

Either way you can't just put "skip it temporarily" with no follow-up, there must be a bug filed, else this will never be fixed.

Reply:
@andrewboie Sorry, I had tried to find the root cause for this couple days but still not know exactly why. It failed due to the fatal error reason is "". The address of <__gcov0.transplanted_function> inside transplanted_function( ) is not correct. It should be like 0xbfffb000,  not should be like 0xc0022dbb. 

=== pass log in qemu_x86 with coverage or qemu_x86_64
START - test_z_mem_map_exec
E: Page fault at address 0xbfffb000 (error code 0x11)
E: Access violation: supervisor thread not allowed to execute

001006be <transplanted_function>:
  1006be:       55                      push   %ebp
  1006bf:       89 e5                   mov    %esp,%ebp 
  1006c1:       a1 18 85 12 00          mov    0x128518,%eax   <=   correct address 0xbfffb000
  1006c6:       8b 15 1c 85 12 00       mov    0x12851c,%edx
  1006cc:       83 c0 01                add    $0x1,%eax


===  fail log in qemu_x86_64 with coverage
START - test_z_mem_map_exec
E: Page fault at address 0xc0022dbb (error code 0x0)  
E: Linear address not present in page tables
E: Access violation: supervisor thread not allowed to read

0000000000100265 <transplanted_function>:
  100265:       55                      push   %rbp
  100266:       48 89 e5                mov    %rsp,%rbp
  100269:       48 89 7d f8             mov    %rdi,-0x8(%rbp)
  10026d:       48 8b 05 ac ad 02 00    mov    0x2adac(%rip),%rax        # 12b020 <__gcov0.transplanted_function> <=   wrong address 0xc0022dbb.

I guess this seems like some thing related to QEMU. And I will keep digging on this.  Thanks for your review and comment!

Reply:
Please at least ensure a bug is filed. Thanks!

Reply:
@andrewboie Ok, got it. Now I understand to issue a bug first when when met this sort of situation. Thanks for your reminder!